File: E-mart_.Net/Backend/Models/Cartitem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace E_mart.Models;

[Table("cartitems")]
[Index("Cartid", Name = "FK5vi0udj5ia7o0h6k6r0op1do6")]
[Index("Productid", Name = "FKp74gufw7pdc246iolcggngpp")]
public partial class Cartitem
{
    [Key]
    [Column("cartitemid")]
    public int Cartitemid { get; set; }

    [Column("added_at")]
    [MaxLength(6)]
    public DateTime? AddedAt { get; set; }

    [Column("quantity")]
    public int Quantity { get; set; }

    [Column("cartid")]
    public int Cartid { get; set; }

    [Column("productid")]
    public int Productid { get; set; }

    [ForeignKey("Cartid")]
    [InverseProperty("Cartitems")]
    public virtual Cart Cart { get; set; } = null!;

    [ForeignKey("Productid")]
    [InverseProperty("Cartitems")]
    public virtual Product Product { get; set; } = null!;
}

File: E-mart_java/Backend/src/main/java/com/example/Models/CartItem.java
package com.example.Models;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "cartitems")
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cartitemid", nullable = false, unique = true)
    private int id;

    @ManyToOne
    @JoinColumn(name = "cartid", nullable = false)
    private Cart cart;

    @ManyToOne
    @JoinColumn(name = "productid", nullable = false)
    private Product product;

    private int quantity;
    private LocalDateTime addedAt;
    
    
//    // Getters and Setters
//	public int getCartItemid() {
//		return cartItemid;
//	}
//	public void setCartItemid(int cartItemid) {
//		this.cartItemid = cartItemid;
//	}
    
    
	public Cart getCart() {
		return cart;
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public void setCart(Cart cart) {
		this.cart = cart;
	}
	public Product getProduct() {
		return product;
	}
	public void setProduct(Product product) {
		this.product = product;
	}
	public int getQuantity() {
		return quantity;
	}
	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}
	public LocalDateTime getAddedAt() {
		return addedAt;
	}
	public void setAddedAt(LocalDateTime addedAt) {
		this.addedAt = addedAt;
	}

 
    
    
}